### PR TITLE
Addressed the recommended changes and used @Valid annotation for validation.

### PR DESCRIPTION
**Pull request by shakibkhan0101**
Using @Valid annotation for validation made code less bulky and light weight, also more readable.

### CODE REVIEW

File: src/main/java/com/globalitgeeks/examninja/usermanagement/dto/ErrorResponse.java [hunk 1]
```diff
@@ -2,11 +2,17 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
 public class ErrorResponse {
-    private String error;
-    public ErrorResponse(String error){
-        this.error=error;
+    private String message;
+    private Map<String, String> error;
+
+    public ErrorResponse(String message) {
```
Comment (ksharmatechie) [inline src/main/java/com/globalitgeeks/examninja/usermanagement/dto/ErrorResponse.java:15]: You need to remove this constructor now.

Comment (shakibkhan0101) [reply src/main/java/com/globalitgeeks/examninja/usermanagement/dto/ErrorResponse.java:15]: Addressed 
Removing the single arg constructor may break the code for two custom business exceptions- IncorrectPassword for login and UserNotFoud for login and change password api,.

**Comment by shakibkhan0101** [comment]
As of now it is up to date

**Comment by shakibkhan0101** [comment]
updated

**Pull request merged**